src/lib.rs

```diff
@@ -334,8 +334,10 @@ impl<Types: IpfsTypes> Ipfs<Types> {
     }
 
     /// Remove block from the ipfs repo.
-    pub async fn remove_block(&self, cid: &Cid) -> Result<(), Error> {
-        Ok(self.repo.remove_block(cid).await?)
+    pub async fn remove_block(&self, cid: Cid) -> Result<Cid, Error> {
+        let cid = cid.clone();
+        self.repo.remove_block(&cid).await?;
+        Ok(cid)
```

Comment:
Unsure if I should be just returning the cid here instead of passing it all the way down and returning it all the way back up, or some mix of both, or if it makes a big difference. I was also thinking this could take a `Cid` and return a `RemovedCid`

Reply:
This needs to account for cid upgrade I think, you'll have that available on rebase.

The clone itself again I think redundant, clippy probably warned on that already? I think the incoming cid should be taken as a reference, so removing it isn't such of a big deal. If the caller wants to continue using their now removed cid they can. If we want it for an json rendering use case we can just use our orginal value for that.

Reply:
Aah not here, this was the `Ipfs`, but in the `Repo`